bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDCyclicTransposeFlowGraphFinder.java

```diff
@@ -0,0 +1,201 @@
+                    .getDestinationNode()
+                    .getEntityName());
+
+            previousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName());
+
+            List<DFDVertex> finalVertices = vertices;
+            vertices = (incomingFlowsToPin.stream()
+                    .flatMap(flow -> handleIncomingFlow(flow, inputPin, finalVertices, sourceNodes, previousNodesInTransposeFlow).stream())
+                    .toList());
+        }
+        return vertices;
+    }
+
+    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> inputPins, List<Node> sourceNodes, List<String> previousNodesInTransposeFlow) {
+        List<DFDVertex> vertices = new ArrayList<>();
+        vertices.add(sink);
+
+        if (sourceNodes.contains(sink.getReferencedElement())) {
+            return vertices;
+        }
+
+        for (Pin inputPin : inputPins) {
+            var copyPreviousNodesInTransposeFlow = new ArrayList<>(previousNodesInTransposeFlow);
+
+            List<Flow> incomingFlowsToPin = dataFlowDiagram.getFlows()
+                    .stream()
+                    .filter(flow -> flow.getDestinationPin()
+                            .equals(inputPin))
+                    .toList();
+
+            if (!loopCheck(previousNodesInTransposeFlow, incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName())) {
+                // better solution needed because to many prints
+                if (!hasCycles) {
+                    logger.warn("Resolving cycles: Stoping cyclic behavior for analysis, may cause unwanted behavior");
+                    hasCycles = true;
+                }
+                continue;
+            }
+
+            copyPreviousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName());
+
+            List<DFDVertex> finalVertices = vertices;
+            vertices = (incomingFlowsToPin.stream()
+                    .flatMap(flow -> handleIncomingFlow(flow, inputPin, finalVertices, sourceNodes, copyPreviousNodesInTransposeFlow).stream())
+                    .toList());
+        }
+        return vertices;
+    }
+
+    private List<DFDVertex> handleIncomingFlow(Flow incomingFlow, Pin inputPin, List<DFDVertex> finalVertices, List<Node> sourceNodes,
+            List<String> previousNodesInTransposeFlow) {
+        List<DFDVertex> result = new ArrayList<>();
+
+        Node previousNode = incomingFlow.getSourceNode();
+
+        List<Pin> previousNodeInputPins = getAllPreviousNodeInputPins(previousNode, incomingFlow);
+
+        List<DFDVertex> previousNodeVertices = determineSinks(new DFDVertex(previousNode, new HashMap<>(), new HashMap<>()), previousNodeInputPins,
+                sourceNodes, previousNodesInTransposeFlow);
+
+        // can we optimize this?
```

Comment:
Is this still relevant? If so, move it into a documenting comment of the method and describe the reason.
Otherwise, remove it

Reply:
no resolved in our conversation during meeting